docs/repo/layout.md

```diff
@@ -28,59 +32,124 @@ These crates are related to the database.
+
+#### Protocol
 
-- [`net/network`](../../crates/net/network): The main P2P networking crate, handling message egress, message ingress, peer management, and session management.
 - [`net/eth-wire`](../../crates/net/eth-wire): Implements the `eth` wire protocol and the RLPx networking stack.
-- [`net/discv4`](../../crates/net/discv4): An implementation of the [discv4][discv4] protocol
-- [`net/ipc`](../../crates/net/ipc): IPC server and client implementation for [`jsonrpsee`][jsonrpsee].
 - [`net/ecies`](../../crates/net/ecies): Implementation of the Elliptic Curve Integrated Encryption Scheme used in the RLPx handshake.
 
-#### RPC
+#### Downloaders
 
-- [`net/rpc-api`](../../crates/rpc/rpc-api): RPC traits
-  - Supported transports: HTTP, WS, IPC
-  - Supported namespaces: `eth_`, `engine_`, `debug_`
-- [`net/rpc`](../../crates/rpc/rpc): Implementation of all ETH JSON RPC traits defined in `rpc-api`.
-- [`net/rpc-types`](../../crates/rpc/rpc-types): Types relevant for the RPC endpoints above, grouped by namespace
+- [`net/downloaders`](../../crates/net/downloaders): Traits defining block body and header downloaders, as well as P2P implementations of both.
 
-#### Downloaders
+### Consensus
+
+Different consensus mechanisms.
+
+- [`consensus/common`](../../crates/consensus/common): Common consensus functions and traits (e.g. fee calculation)
+- [`consensus/auto-seal`](../../crates/consensus/auto-seal): A consensus mechanism that auto-seals blocks for local development (also commonly known as "auto-mine")
+- [`consensus/beacon`](../../crates/consensus/beacon): Consensus mechanism that handles messages from a beacon node ("eth2")
 
-- [`net/downloaders`](../../crates/net/downloaders/): Block body and header downloading strategies.
+### Execution
 
-### Ethereum
+Crates related to transaction execution.
 
-These crates are Ethereum-specific (e.g. EVM, consensus, transaction pools).
+- [`revm`](../../crates/revm): An implementation of an executor using `revm`
+- [`revm/revm-inspectors`](../../crates/revm/revm-inspectors): Various revm inspectors for debugging and building traces for the trace-related RPC APIs.
+- [`revm/revm-primitives`](../../crates/revm/revm-primitives): A crate for handling revm-reth type conversion
+
+### Sync
+
+These crates implement the main syncing drivers of reth.
+
+- [`blockchain-tree`](../../crates/blockchain-tree): A tree-like structure for handling multiple chains of unfinalized blocks. This is the main component during live sync (i.e. syncing at the tip)
+- [`stages`](../../crates/stages): A pipelined sync, including implementation of various stages. This is used during initial sync and is faster than the tree-like structure for longer sync ranges.
+- [`staged-sync`](../../crates/staged-sync): A catch-all for various things currently, to be removed
+
+### RPC
+
+Crates related to the RPC component (including JSONRPC and IPC transports)
```

Comment:
perhaps not even worth mentioning here since also listed under transports

```suggestion
Crates related to the RPC component (including IPC transport)
```